mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/PuzzleGameScreen.kt

```diff
@@ -5,39 +5,116 @@ import androidx.compose.foundation.rememberScrollState
 import androidx.compose.foundation.verticalScroll
 import androidx.compose.material.*
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.BlackKing
+import ch.epfl.sdp.mobile.ui.ChessIcons
 import ch.epfl.sdp.mobile.ui.game.*
+import ch.epfl.sdp.mobile.ui.game.classic.ClassicChessBoard
+import ch.epfl.sdp.mobile.ui.puzzles.PuzzleInfoState.*
 
 /**
  * This screen display an ongoing chess puzzle
  *
  * @param state the [GameScreenState] that manage the composable contents
  * @param modifier the [Modifier] for the composable
  * @param contentPadding the [PaddingValues] for this composable.
+ * @param snackbarHostState the [SnackbarHostState] for this composable.
  */
 @Composable
-fun PuzzleGameScreen(
-    state: PuzzleGameScreenState,
+fun <Piece : ChessBoardState.Piece> PuzzleGameScreen(
+    state: PuzzleGameScreenState<Piece>,
     modifier: Modifier = Modifier,
     contentPadding: PaddingValues = PaddingValues(),
+    snackbarHostState: SnackbarHostState = remember { SnackbarHostState() },
 ) {
+  val strings = LocalLocalizedStrings.current
+  val typo = MaterialTheme.typography
+  val firstColor =
+      when (state.puzzleInfo.playerColor) {
+        ChessBoardState.Color.White -> ChessBoardState.Color.Black
+        ChessBoardState.Color.Black -> ChessBoardState.Color.White
+      }
   Scaffold(
       modifier = modifier,
+      scaffoldState = rememberScaffoldState(snackbarHostState = snackbarHostState),
+      topBar = {
+        GameScreenTopBar(
+            onBackClick = state::onBackClick,
+            onArClick = {},
+            onListenClick = state::onListenClick,
+            listening = state.listening,
+            modifier = Modifier.fillMaxWidth(),
+        )
+      },
       content = { scaffoldPadding ->
         Column(
             modifier =
                 modifier
                     .verticalScroll(rememberScrollState())
                     .padding(contentPadding)
                     .padding(scaffoldPadding)
-                    .padding(start = 16.dp, end = 16.dp, top = 48.dp, bottom = 48.dp),
-            verticalArrangement = Arrangement.spacedBy(32.dp),
+                    .padding(start = 32.dp, end = 32.dp, top = 48.dp, bottom = 48.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
         ) {
-          Text("Puzzle id: ${state.puzzle.uid}")
-          Text("Playing as: ${state.puzzle.playerColor}")
-          Text("Elo: ${state.puzzle.elo}")
+          Text(strings.puzzlesTitle.uppercase(), style = typo.h6)
+          PuzzleDirective(color = state.puzzleInfo.playerColor, puzzleState = state.puzzleState)
+          ProvideTextStyle(typo.subtitle1) { ClassicChessBoard(state) }
+          Spacer(Modifier)
```

Comment:
`Spacer(Modifier)` will create an empty `Layout`, so in this case you'll end up visually applying the vertical arrangement spacing of 16.dp twice. Is this the behaviour you're expecting ?

If that's the goal, I'd suggest going for a `Modifier.padding(top = 16.dp)` applied to the next element, which is less error-prone.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/309/commits/be1fccf58cc569963212f8559919a64d3e5f73cd